### PR TITLE
MAINT: use request payload instead of multiple path args

### LINKED ISSUE #34: Use a request payload rather than multiple path args
https://github.com/eigenfoo/euphoria/blob/master/euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java#L31

Style-wise, creating a new db object would usually be from a request payload.

See https://github.com/dhvanilshah/Symmetric-Social-Network-ECE366/issues/5 for an example

### PR DESCRIPTION
**Pull request by eigenfoo**
Closes #34.

### CODE REVIEW

**Review by wside — DISMISSED** [review]

**Comment by eigenfoo** [comment]
Pending another PR by @lipovac before we merge this: the frontend needs to be able to send the appropriate http requests.

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java [hunk 1]
```diff
@@ -203,12 +200,13 @@ public PostingHandlers(final ObjectMapper objectMapper, final Config config) {
...
         try {
-            Integer companyId = Integer.valueOf(rc.pathArgs().get("companyId"));
-            String jobTitle = rc.pathArgs().get("jobTitle");
-            String description = rc.pathArgs().get("description");
-            Location location = Location.valueOf(rc.pathArgs().get("location"));
-            Industry industry = Industry.valueOf(rc.pathArgs().get("industry"));
-            SkillLevel skillLevel = SkillLevel.valueOf(rc.pathArgs().get("skillLevel"));
+            Map jsonMap = objectMapper.readValue(rc.request().payload().get().toByteArray(), Map.class);
+            Integer companyId = (Integer) jsonMap.get("companyId");
+            String jobTitle = jsonMap.get("companyId").toString();
```
Comment (wside) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java]: Typo: should be "jobTitle" on line 205

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java [hunk 2]
```diff
@@ -203,12 +200,13 @@ public PostingHandlers(final ObjectMapper objectMapper, final Config config) {
...
     public List<Posting> createPosting(final RequestContext rc) {
         try {
-            Integer companyId = Integer.valueOf(rc.pathArgs().get("companyId"));
-            String jobTitle = rc.pathArgs().get("jobTitle");
-            String description = rc.pathArgs().get("description");
-            Location location = Location.valueOf(rc.pathArgs().get("location"));
-            Industry industry = Industry.valueOf(rc.pathArgs().get("industry"));
-            SkillLevel skillLevel = SkillLevel.valueOf(rc.pathArgs().get("skillLevel"));
+            Map jsonMap = objectMapper.readValue(rc.request().payload().get().toByteArray(), Map.class);
+            Integer companyId = (Integer) jsonMap.get("companyId");
```
Comment (wside) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java]: change to 
Integer companyId = Integer.parseInt(jsonMap.get("companyId").toString());

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java [hunk 3]
```diff
@@ -28,17 +28,13 @@ public PostingHandlers(final ObjectMapper objectMapper, final Config config) {
...
-                Route.sync("GET", "/posting/getRandom", this::getRandomPostings),
-                Route.sync("POST",
-                        "/posting/<companyId>/<jobTitle>/<description>/<location>/<industry>/<skillLevel>",
-                        this::createPosting),
-                Route.sync("PUT",
-                        "/posting/<postingId>/<jobTitle>/<description>/<location>/<industry>/<skillLevel>",
-                        this::editPosting),
-                Route.sync("DELETE", "/posting/<postingId>", this::deletePosting)
+                Route.sync("GET", "/api/posting/<postingId>", this::getPosting),
+                Route.sync("GET", "/api/posting/all", this::getAllPostings),
```
Comment (wside) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java:32]: need a line for getRandomPostings somewhere in here

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java (hunk 1 above)
Comment (eigenfoo) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java]: Good catch!

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java (hunk 2 above)
Comment (eigenfoo) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/PostingHandlers.java]: :+1:

**Review by wside — APPROVED** [review]

**Pull request merged**